Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1

```diff
@@ -0,0 +1,349 @@
+    {
+        Write-Verbose -Message "The IPAddressAllowList is not configured, for that the IPAddressEnforcement parameter can not be set and will be removed"
+        $CurrentParameters.Remove("IPAddressEnforcement")
+        $CurrentParameters.Remove("IPAddressAllowList")
+    }
+    $tenant = Set-PnPTenant @CurrentParameters
+}
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param (
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $DisplayStartASiteOption,
+
+        [Parameter()]
+        [System.String]
+        $StartASiteFormUrl,
+
+        [Parameter()]
+        [System.Boolean]
+        $IPAddressEnforcement,
+
+        [Parameter()]
+        [System.String]
+        $IPAddressAllowList,
+
+        [Parameter()]
+        [System.UInt32]
+        $IPAddressWACTokenLifetime,
+
+        [Parameter()]
+        [System.Boolean]
+        $CommentsOnSitePagesDisabled,
+
+        [Parameter()]
+        [System.Boolean]
+        $SocialBarOnSitePagesDisabled,
+
+        [Parameter()]
+        [System.Boolean]
+        $DisallowInfectedFileDownload,
+
+        [Parameter()]
+        [System.boolean]
+        $ExternalServicesEnabled,
+
+        [Parameter()]
+        [System.Boolean]
+        $EmailAttestationRequired,
+
+        [Parameter()]
+        [System.UInt32]
+        $EmailAttestationReAuthDays,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $CentralAdminUrl,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+    Test-PnPOnlineConnection -SPOCentralAdminUrl $CentralAdminUrl -GlobalAdminAccount $GlobalAdminAccount
+
+    Write-Verbose -Message "Testing SPO Tenant"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+    return Test-Office365DSCParameterState -CurrentValues $CurrentValues `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck @("IsSingleInstance", `
+            "CentralAdminUrl", `
+            "GlobalAdminAccount", `
+            "DisplayStartASiteOption", `
+            "StartASiteFormUrl", `
+            "IPAddressEnforcement", `
+            "IPAddressAllowList", `
+            "IPAddressWACTokenLifetime", `
+            "CommentsOnSitePagesDisabled", `
+            "SocialBarOnSitePagesDisabled", `
+            "DisallowInfectedFileDownload", `
+            "ExternalServicesEnabled", `
+            "EmailAttestationRequired", `
+            "EmailAttestationReAuthDays"
+    )
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $DisplayStartASiteOption,
```

Comment:
Only the mandatory parameters are required in this signature. To be consistent with other resources please only include IsSingleInstance, GlobalAdminAccount and CentralAdminUrl